qiskit_experiments/calibration_management/save_utils.py

```diff
@@ -0,0 +1,253 @@
+
+    """
+
+    param_name: str
+    qubits: List[int]
+    schedule: str = ""
+    value: float = 0.0
+    datetime: datetime = datetime.now()
+    valid: bool = True
+    exp_id: str = ""
+    group: str = ""
+
+    @classmethod
+    def apply_schema(cls, data: Dict[str, Any]):
+        """Consume dictionary and returns canonical data model."""
+        return ParameterModelV1(**data)
+
+
+@dataclass(frozen=True)
+class ScheduleModelV1:
+    """A data schema of a single templated schedule.
+
+    .. note::
+        This is intentionally agnostic to the data structure of
+        Qiskit Experiments Calibrations for portability.
+
+    """
+
+    data: ScheduleBlock
+    qubits: List[int] = field(default_factory=list)
+    num_qubits: int = 0
+
+    @classmethod
+    def apply_schema(cls, data: Dict[str, Any]):
+        """Consume dictionary and returns canonical data model."""
+        return ScheduleModelV1(**data)
+
+
+@dataclass
+class CalibrationModelV1:
+    """A data schema of whole device representation.
+
+    .. note::
+        This is intentionally agnostic to the data structure of
+        Qiskit Experiments Calibrations for portability.
+
+    """
+
+    backend_name: str
+    backend_version: str
+    device_coupling_graph: List[List[int]]
+    control_channel_map: ControlChannelMap
+    schedules: List[ScheduleModelV1] = field(default_factory=list)
+    parameters: List[ParameterModelV1] = field(default_factory=list)
+    schema_version: str = "1.0"
+
+    @classmethod
+    def apply_schema(cls, data: Dict[str, Any]):
+        """Consume dictionary and returns canonical data model."""
+        in_data = {}
+        for key, value in data.items():
+            if key == "schedules":
+                value = list(map(ScheduleModelV1.apply_schema, value))
+            if key == "parameters":
+                value = list(map(ParameterModelV1.apply_schema, value))
+            in_data[key] = value
+        return CalibrationModelV1(**in_data)
+
+
+def to_dict(
+    cals: Calibrations,
+    most_recent_only: bool = True,
+) -> Dict[str, Any]:
+    """A helper function to convert calibration data into dictionary.
+
+    Args:
+        cals: A calibration instance to save.
+        most_recent_only: Set True to save calibration parameters with most recent time stamps.
+
+    Returns:
+        Canonicalized calibration data in dictionary format.
+    """
+    # This can dispatch canonicalize function as version evolves.
+    model = _canonicalize_calibration_data_v1(cals, most_recent_only)
+    return asdict(model)
+
+
+def from_dict(
+    cal_data: Dict[str, Any],
+) -> Calibrations:
+    """A helper function to build calibration instance from canonical dictionary.
+
+    Args:
+        cal_data: Calibration data that conforms to the predefined data schema.
+
+    Returns:
+        Calibration instance.
+
+    Raises:
+        ValueError: When input data model version is no longer supported.
```

Comment:
```suggestion
        ValueError: When input data model version is no longer supported.
```
Are you sure you mean `no longer supported`? Reads like this will change in the future?